src/coreclr/vm/appdomain.cpp

```diff
@@ -1480,7 +1474,6 @@ void AppDomain::Create()
 
     NewHolder<AppDomain> pDomain(new AppDomain());
     pDomain->Init();
-    pDomain->SetStage(AppDomain::STAGE_OPEN);
```

Comment:
Notice that the stage is set to STAGE_OPEN before the AppDomain is published to m_pTheAppDomain that makes it visible to the rest of the system. It means that the only stage that the rest of system can observe is STAGE_OPEN.